receiver/receivertest/nop_logs_receiver.go

```diff
@@ -0,0 +1,54 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package receivertest // import "go.opentelemetry.io/collector/receiver/receivertest"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+var nopLogsComponentType = component.MustNewType("nop_logs")
```

Comment:
Love the functional options pattern here to avoid creating an entirely new receiver!

> It's a bit awkward with because we want to keep the defaults as true, but here's what I came up with to do it:

What if we negated the options so they're `WithoutTraces()`, `WithoutMetrics()`, and `WithoutLogs()`? That would let us preserve the current behavior as the default. Not sure if the `Without*` is more confusing than `With*`, though?

Reply:
@djaglowski I implemented your idea, albeit with `Without*` options (see previous comment) in 3fe135652.  I also had to dynamically name the receiver's type based on the options to avoid a collision with the default `nop` type.  Let me know what you think.  Thanks!

Reply:
Hmm, having to modify the component type is not ideal. _Maybe_ it's a little nicer if we make the options mutually exclusive e.g. `WithOnly(component.DataType)` because then at last the alternate types can just be `noplogs` / `nopmetrics`, etc.

Either way, this personally still seems better than defining multiple components but it will be up to a maintainer.

Reply:
> Hmm, having to modify the component type is not ideal. _Maybe_ it's a little nicer if we make the options mutually exclusive e.g. `WithOnly(component.DataType)` because then at last the alternate types can just be `noplogs` / `nopmetrics`, etc.

With this approach, I think the functional options pattern makes less sense, at least ergonomically, if the caller is required to pass at most one `WithOnly(component.DataType)` option to the factory?

Perhaps it would make sense to leave the `NewNopFactory` implementation as-is, that is, to return a factory that constructs `nop` receivers supporting all data types.  And then introduce something like `NewNopFactoryForType(component.DataType)` that returns a factory that constructs `nop` receivers supporting only the given data type? I implemented this in ce1ad7d5c, if you want to take a look.

> Either way, this personally still seems better than defining multiple components but it will be up to a maintainer.

Agreed.

Reply:
> NewNopFactoryForType

This looks better to me. Thanks for iterating on it.